### ISSUE TITLE
Istio sidecar init-container requires to run as `root`, specify it explicitly in the ConfigMap

### ISSUE THREAD

**Issue by shakti-das** [istio/istio#10358]
**Describe the bug**
Istio sidecar init-container requires to run as `root` because of `iptables`. It does not specify it explicitly in the sidecar injector ConfigMap. In our environment, Pods by default run as a non-root user. So, the init-container inherits:
```
  securityContext:
    fsGroup: <int>
    runAsNonRoot: true
    runAsUser: <int>
```
We need to explicitly override the init-container as:
```
  securityContext:
    runAsNonRoot: true
    runAsUser: 0
```

**Expected behavior**
The `runAsUser: 0` configuration is currently not present in the istio-sidecar-configmap by default. Would expect this to be present by default since it's required to run `iptables`

**Steps to reproduce the bug**
1. Set [PodSecurityContext](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.12/#podsecuritypolicyspec-v1beta1-extensions) with `runAsUser` different than root.
1. Try the sidecar auto-injection. The auto-injected init-container fails to start with error:
```
+ iptables -t nat -N ISTIO_REDIRECT
iptables v1.6.0: can't initialize iptables table `nat': Permission denied (you must be root)
Perhaps iptables or your kernel needs to be upgraded.
+ dump
+ iptables-save
iptables-save v1.6.0: Cannot initialize: Permission denied (you must be root)
```

Even if `privileged=true` is set at container level SecurityContext, the issue is reproducible.

**Version**
```
$ istioctl version
Version: 1.0.2
GitRevision: d639408fded355fb906ef2a1f9e8ffddc24c3d64
User: root@66ce69d4a51e
Hub: gcr.io/istio-release
GolangVersion: go1.10.1
BuildStatus: Clean

$ kubectl version
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.1", GitCommit:"d4ab47518836c750f9949b9e0d387f20fb92260b", GitTreeState:"clean", BuildDate:"2018-04-13T22:27:55Z", GoVersion:"go1.9.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.7", GitCommit:"dd5e1a2978fd0b97d9b78e1564398aeea7e7fe92", GitTreeState:"clean", BuildDate:"2018-04-18T23:58:35Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}

```

**Installation**
[Istio Minimal Installation steps were followed](https://istio.io/docs/setup/kubernetes/minimal-install/)

**Environment**
Own k8s cluster.

**Comment by shakti-das** [comment]
I have local changes for this and can send out a PR to set these defaults if deemed useful.

**Comment by jaredallard** [comment]
@shakti-das I'd be interested in seeing how this is fixed

**Comment by linsun** [comment]
what k8s env are you using?  is it open shift?

It is known that in some k8s distro the security is tightened thus root privilege is required for init container.  The community is aware of this and are working towards a solution using https://github.com/istio/cni

**Comment by shakti-das** [comment]
@linsun We have our own k8s deployment and the restrictions are intended restrictions. But we have a workaround for now till the CNI solution is ready. So, I think this issue can be closed.

**Comment by linsun** [comment]
Thank you @shakti-das !

**Comment by MBauer83** [comment]
I have attempted to get redis, etcd, elasticsearch and mariadb clusters running on Azure AKS with istio in versions 1.0.5, 1.1.0-snapshot.4 & 1.1.0-snapshot.5, and have not managed to get either working with sidecar-injection active. The pods fail healthchecks, crash or simply cannot communicate. 

This is irrespective of using CNI or not, irrespective of sidecar-injection-configmap having runAsUser: 0 and irrespective of application pod security configurations and fsGroup configurations. I have also tried with or without mtls enabled (and with or without permissive policy when using mtls). None of the workarounds, nor the intended cni solution seem to be working. 

This issue here seemed to be the most general one related to my problems - should I comment on related issues for redis, elastic, mariadb/mysql (etc - or their respective helm installs) with my experiences of the workarounds and cni not working, or is there a better way to track this?

**Comment by sdake** [comment]
> I have attempted to get redis, etcd, elasticsearch and mariadb clusters running on Azure AKS with istio in versions 1.0.5, 1.1.0-snapshot.4 & 1.1.0-snapshot.5, and have not managed to get either working with sidecar-injection active. The pods fail healthchecks, crash or simply cannot communicate.
> 
> This is irrespective of using CNI or not, irrespective of sidecar-injection-configmap having runAsUser: 0 and irrespective of application pod security configurations and fsGroup configurations. I have also tried with or without mtls enabled (and with or without permissive policy when using mtls). None of the workarounds, nor the intended cni solution seem to be working.
> 
> This issue here seemed to be the most general one related to my problems - should I comment on related issues for redis, elastic, mariadb/mysql (etc - or their respective helm installs) with my experiences of the workarounds and cni not working, or is there a better way to track this?

@MBauer83 probably bestto open a new issue as this one is closed and won't be examined.

**Comment by jaredallard** [comment]
Not sure that this should be closed, especially since a fix PR is open and awaiting feedback / a merge. A workaround doesn't always suffice. I used to be able to edit it manually, but now GCP's istio integration replaces all modifications to the configmap, thus rendering that workaround broken.

**Comment by kbruner** [comment]
Also hitting this issue with the istio-init sidecar when it's injected into a deployment with a strict pod-level securityContext.

**Comment by philrud** [comment]
This was fixed couple weeks ago with this PR: https://github.com/istio/istio/pull/12708 (already merged into release-1.1 branch, should go live with 1.1.3).
Specifically to fix Redis charts (by Bitnami), sorry haven't seen this issue/PR before. 

cc @duderino

**Comment by howardjohn** [comment]
Fix is merged in and will be released in 1.1.3 (expected around 4/15). Thanks everyone

**Comment by Tushar120120** [comment]
Does it mean httpbin example will not work in 1.1.2?
